pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -349,6 +352,100 @@ contract ManagedPool is BaseWeightedPool, ReentrancyGuard {
         emit SwapEnabledSet(swapEnabled);
     }
 
+    /**
+     * @dev Remove a token from the pool. This is a permissioned function that performs the following operations:
+     * - Verify there are enough tokens, and there is no ongoing weight change
+     * - Calculate the BPT value of the full token balance, and returns it for possible use by the caller
+     * - Exit the pool, returning the full balance of the token to the recipient
+     * - Deregister the token in the Vault (since it now has zero balance)
+     * - Reduce the total weight by the weight of the token removed, and update the number of tokens and `_tokenState`
+     *   (all other weights then scale accordingly)
+     */
+    function removeToken(uint256 tokenIndex, address recipient) external authenticate whenNotPaused returns (uint256) {
+        return _removeToken(tokenIndex, recipient);
+    }
+
+    function _removeToken(uint256 tokenIndex, address recipient) internal returns (uint256) {
+        // Calculate the BPT value of the full token balance (other values returned to avoid duplication)
+        //(uint256 bptAmountIn, uint256 minAmountOut, uint256 normalizedWeight) = getTotalBptForToken(tokenIndex);
+        // Exit the pool, returning the full balance of the token to the recipient
+        (IERC20[] memory tokens, uint256[] memory rawBalances, ) = getVault().getPoolTokens(getPoolId());
+        _require(tokenIndex < tokens.length, Errors.OUT_OF_BOUNDS);
+        _require(tokens.length > 2, Errors.MIN_TOKENS);
+        // Verify there is no ongoing weight change
+        _require(
+            0 == _calculateWeightChangeProgress() || FixedPoint.ONE == _calculateWeightChangeProgress(),
+            Errors.REMOVE_TOKEN_DURING_WEIGHT_CHANGE
+        );
```

Comment:
```suggestion
        {
            uint256 weightChangeProgress = _calculateWeightChangeProgress();
            _require(weightChangeProgress == 0 || weightChangeProgress == FixedPoint.ONE);
        }
```

Reply:
Can we remove a token if there is a scheduled weight change in the future?

Reply:
Technically yes, at the pool level, since it's happening immediately and won't interfere. There could be a rebalancing or something involving other tokens that you might still want to happen.

They could always just issue another updateWeightsGradually to stop it if they forgot or it was screwing up; it would be safer to prevent it, though.

At the controller level, if the remove is happening after a delay (TBD where the delays are; maybe they're only on adds), maybe the controller would want to block scheduling a removal if there was a future weight change: and similarly block calls to updateWeightsGradually if there was a pending removal.